cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -259,72 +271,135 @@ void accumulate_vertex_results(
                       });
   }
 
-  edge_src_property_t<vertex_t, thrust::tuple<vertex_t, edge_t, weight_t>> src_properties(
-    handle, graph_view);
-  edge_dst_property_t<vertex_t, thrust::tuple<vertex_t, edge_t, weight_t>> dst_properties(
-    handle, graph_view);
+  edge_src_property_t<vertex_t, vertex_t> src_distances(handle, graph_view);
+  edge_src_property_t<vertex_t, edge_t> src_sigmas(handle, graph_view);
+  edge_src_property_t<vertex_t, weight_t> src_deltas(handle, graph_view);
+  edge_dst_property_t<vertex_t, vertex_t> dst_distances(handle, graph_view);
+  edge_dst_property_t<vertex_t, edge_t> dst_sigmas(handle, graph_view);
+  edge_dst_property_t<vertex_t, weight_t> dst_deltas(handle, graph_view);
+
+  // Update distances and sigmas initially (deltas start as 0)
+  update_edge_src_property(handle, graph_view, distances.begin(), src_distances.mutable_view());
+  update_edge_src_property(handle, graph_view, sigmas.begin(), src_sigmas.mutable_view());
+  update_edge_dst_property(handle, graph_view, distances.begin(), dst_distances.mutable_view());
+  update_edge_dst_property(handle, graph_view, sigmas.begin(), dst_sigmas.mutable_view());
+  update_edge_src_property(handle, graph_view, deltas.begin(), src_deltas.mutable_view());
+  update_edge_dst_property(handle, graph_view, deltas.begin(), dst_deltas.mutable_view());
```

Comment:
You can use view_concat here as well, so instead of making 6 primitive calls, it can be just 2.